hyperspy/docstrings/plot.py

```diff
@@ -88,20 +88,65 @@
 
         Parameters
         ----------
-        %s""" % BASE_PLOT_DOCSTRING_PARAMETERS
+        """
+
 
 PLOT1D_DOCSTRING = \
-"""norm : str, optional
+    """norm : str, optional
             The function used to normalize the data prior to plotting.
             Allowable strings are: ``'auto'``, ``'linear'``, ``'log'``.
             (default value is ``'auto'``).
             If ``'auto'``, intensity is plotted on a linear scale except when
             ``power_spectrum=True`` (only for complex signals).
+        autoscale : str
+            Set the axes to be reset to their maxima when the navigation
+            indices or data are changed. If 'x' is in the string, the
+            horizontal axis is reset automatically. If 'z' is in the string,
+            the vertical axis is reset automatically. Default is 'z'.
         """
 
+
 PLOT2D_DOCSTRING = \
     """colorbar : bool, optional
             If true, a colorbar is plotted for non-RGB images.
+        autoscale : str
+            Set the axes to be reset to their maxima when the navigation
+            indices or data are changed. If 'x' or 'y' are in the string, the
+            extent of the image are reset automatically. If 'z' is in the
+            string, the displayed value of the image are reset automatically.
```

Comment:
Is this docstring clear enough?
There is a typo here: displayed value*s* but since there is a bit a queue on travis and azure pipeline, I will fix it after this is reviewed.

Reply:
What about:

The string must contain any combination of the 'x', 'y' and 'z' characters. If 'x' or 'y' are in the string, the corresponding axis limits are set to cover the full range of the data at a given position. If 'z' is in the string when plotting images, the contrast will cover the full range of intensities of the image at each position.

Reply:
With changes regarding `z`:

The string must contain any combination of the 'x', 'y' and 'z' characters. If 'x' or 'y' are in the string, the corresponding axis limits are set to cover the full range of the data at a given position. If 'z' is in the string, the contrast of the image will be set automatically according to `vmin` and `vmax` when the data are changed or navigation indices are changed.

Reply:
A small variation:

The string must contain any combination of the 'x', 'y' and 'z' characters. If 'x' or 'y' are in the string, the corresponding axis limits are set to cover the full range of the data at a given position. If 'z' is in the string, the contrast of the image will be set automatically according to `vmin` and `vmax` when the data or navigation indices change.

Reply:
Yes, it is good!